src/xcode/ENA/ENA/Source/Scenes/AntigenTestProfile/Cells/QRCodeCellViewModel.swift

```diff
@@ -0,0 +1,81 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import Contacts
+
+struct QRCodeCellViewModel {
+
+	// MARK: - Init
+
+	init(
+		antigenTestProfile: AntigenTestProfile,
+		backgroundColor: UIColor,
+		boarderColor: UIColor
+	) {
+		self.antigenTestProfile = antigenTestProfile
+		self.backgroundColor = backgroundColor
+		self.boarderColor = boarderColor
+	}
+
+	// MARK: - Internal
+
+	let backgroundColor: UIColor
+	let boarderColor: UIColor
+
+	// QRCode image with vCard data inside - will create an empty image if data is broken
+	var qrCodeImage: UIImage {
+		guard let vCardString = String(data: vCardData, encoding: .utf8),
+			  let QRCodeImage = UIImage.qrCode(
+				with: vCardString,
+				encoding: .utf8,
+				size: CGSize(width: 280.0, height: 280.0),
+				qrCodeErrorCorrectionLevel: .medium
+			  )
+		else {
+			Log.error("Failed to create QRCode image for vCard data")
+			return UIImage()
+		}
+		return QRCodeImage
+	}
+
+	// create vCard data, CNContactVCardSerialization creates a version 3.0 output
+	var vCardData: Data {
```

Comment:
Is this still needed. Because there is a v4 implementation?

Reply:
yes in this branch - will get removed by the V4 PR